src/plugins/data/server/search/routes/search.ts

```diff
@@ -69,12 +72,22 @@ export function registerSearchRoute(router: DataPluginRouter): void {
                 isStored,
                 isRestore,
                 retrieveResults,
+                stream,
               }
             )
             .pipe(first())
             .toPromise();
 
-          return res.ok({ body: response });
+          if (response && (response.rawResponse as unknown as IncomingMessage).pipe) {
+            return res.ok({
+              body: response.rawResponse,
+              headers: {
+                requestParams: JSON.stringify(response.requestParams),
```

Comment:
Should we be prefixing this with something like `kbn-search-request-params` or something?